dev-client/src/model/permissions/permissions.ts

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {ProjectMembershipProjectRoleChoices} from 'terraso-client-shared/graphqlSchema/graphql';
+import {SiteUserRole} from 'terraso-client-shared/selectors';
+
+export const matchesOne =
+  <T>(cmp: (a: T, b: T) => boolean) =>
+  (examples: T[]) =>
+  (sample: T) =>
+    examples.filter(x => cmp(x, sample)).length > 0;
```

Comment:
Suggestion re: Ruxandra's comment -- According to [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find), I think it's just 
```suggestion
    examples.find(x => cmp(x, sample)) !== undefined;
```
which would make sense to me since we only expect it to match and return one, so it can return once it finds one